src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -275,18 +275,17 @@ static void add_pending_pick(pending_pick** root,
 typedef struct pending_ping {
   struct pending_ping* next;
 
-  /* args for wrapped_notify */
-  wrapped_rr_closure_arg wrapped_notify_arg;
+  /* args for sending the ping */
+  grpc_closure* on_initiate;
+  grpc_closure* on_ack;
 } pending_ping;
 
-static void add_pending_ping(pending_ping** root, grpc_closure* notify) {
+static void add_pending_ping(pending_ping** root, grpc_closure* on_initiate,
+                             grpc_closure* on_ack) {
   pending_ping* pping = (pending_ping*)gpr_zalloc(sizeof(*pping));
-  pping->wrapped_notify_arg.wrapped_closure = notify;
-  pping->wrapped_notify_arg.free_when_done = pping;
+  pping->on_initiate = on_initiate;
```

Comment:
no, don't remove `wrapped_notify_arg`: it's hijacking the original closures because grpclb contains an instance of round robin inside and manages references to the internal round robin instances to make sure it doesn't disappear.

Reply:
Fixed. Added wrappers for `on_initiate` and `on_ack`.

But I'm not sure if wrapping pending pings with `wrapped_notify_arg` is necessary. It looks like the pending pings are using `wrapped_notify_arg` only for referencing the rr policy. In `create_rr_locked()`, if `glb_policy->rr_policy` is still valid at the time we call `grpc_lb_policy_ping_one_locked()`, the pings will be sent to the connected subchannels. After this, we don't have to keep a reference of the rr policy until the pings are done. If the connected subchannels are destroyed while there are in-flight pings, their tranports should be able to cancel these pings properly. So we may not have to reference `glb_policy->rr_policy` for pending pings, and a plain simple `grpc_closure` should be adequate for the ping closures here.

Reply:
Hmm good point. I think you are right.

Reply:
I should probably keep the reference as is in this PR. This PR has already had a few troubles for the import. I'll do the reference change in a follow-up PR.